internal/command/postgres/failover.go

```diff
@@ -128,3 +141,173 @@ func runFailover(ctx context.Context) (err error) {
 	fmt.Fprintf(io.Out, "Failover complete\n")
 	return
 }
+
+func flexFailover(ctx context.Context, machines []*api.Machine, app *api.AppCompact) error {
+	io := iostreams.FromContext(ctx)
+
+	leader, err := pickLeader(ctx, machines)
+	if err != nil {
+		return err
+	}
+
+	fmt.Fprintf(io.Out, "Performing a failover\n")
+
+	primary_region := ""
+	if len(machines) >= 3 {
+		for _, machine := range machines {
+			if region, ok := machine.Config.Env["PRIMARY_REGION"]; ok {
```

Comment:
Since we are already doing this loop, we could construct a new slice that contains only the machines residing within the primary region to pass to the pickNewLeader.   This allows us to short-circuit if the number of in-region machines is less than three, and it'll allow us to skip the primaryRegion checks when selecting a new leader.